Src/ILGPU/IR/Construction/IRRebuilder.cs

```diff
@@ -460,6 +637,31 @@ private void InitPhiValues(in BlockCollection blocks)
                 targetPhiBuilder.Seal();
             }
 
+            // If we did not modify all the blocks, ensure that the IR nodes in these
+            // unmodified blocks are rewired to correctly reference the newly created IR
+            // nodes in the blocks that we did modify.
+            if (hasUnmodifiedBlocks)
+            {
+                var phiRemapper = new PartialBlockRemapper(this);
+
+                foreach (var node in valueMapping)
+                {
+                    var oldNode = node.Key;
+                    var newNode = node.Value;
+
+                    if (oldNode != newNode &&
+                        oldNode.CanBeReplaced &&
```

Comment:
Should we assert that this value can be replaced? Are there any other cases in which this can happen?